spec/v2/providers/storage.spec.ts

```diff
@@ -0,0 +1,336 @@
+      delete process.env.GCLOUD_PROJECT;
+    });
+
+    it('should create a minimal trigger', () => {
+      const result = storage._onOperation(() => 42, 'event-type', 'my-bucket');
+
+      expect(result.__trigger).to.deep.equal({
+        apiVersion: 2,
+        platform: 'gcfv2',
+        labels: {},
+        eventTrigger: EVENT_TRIGGER,
+      });
+    });
+
+    it('should create a complex trigger with appropriate values', () => {
+      const result = storage._onOperation(() => 42, 'event-type', {
+        ...FULL_OPTIONS,
+        bucket: 'my-bucket',
+      });
+
+      expect(result.__trigger).to.deep.equal({
+        ...FULL_TRIGGER,
+        eventTrigger: EVENT_TRIGGER,
+      });
+    });
+
+    it('should merge options and globalOptions', () => {
+      options.setGlobalOptions({
+        concurrency: 20,
+        region: 'europe-west1',
+        minInstances: 1,
+      });
+
+      const result = storage._onOperation(() => 42, 'event-type', {
+        bucket: 'my-bucket',
+        region: 'us-west1',
+        minInstances: 3,
+      });
+
+      expect(result.__trigger).to.deep.equal({
+        apiVersion: 2,
+        platform: 'gcfv2',
+        concurrency: 20,
+        minInstances: 3,
+        regions: ['us-west1'],
+        labels: {},
+        eventTrigger: EVENT_TRIGGER,
+      });
+    });
+  });
+
+  describe('onObjectArchived', () => {
+    it('should accept only handler', () => {
+      const configStub = sinon
+        .stub(config, 'firebaseConfig')
+        .returns({ storageBucket: 'default-bucket' });
+
+      const result = storage.onObjectArchived(() => 42);
+
+      configStub.restore();
+      expect(result.__trigger).to.deep.equal({
+        apiVersion: 2,
+        platform: 'gcfv2',
+        labels: {},
+        eventTrigger: {
+          eventType: 'google.cloud.storage.object.v1.archived',
+          resource: 'default-bucket',
+          service: 'storage.googleapis.com',
+        },
+      });
+    });
+
+    it('should accept bucket and handler', () => {
+      const result = storage.onObjectArchived('my-bucket', () => 42);
+
+      expect(result.__trigger).to.deep.equal({
+        apiVersion: 2,
+        platform: 'gcfv2',
+        labels: {},
+        eventTrigger: {
+          eventType: 'google.cloud.storage.object.v1.archived',
+          resource: 'my-bucket',
+          service: 'storage.googleapis.com',
+        },
+      });
+    });
+
+    it('should accept opts and handler', () => {
+      const result = storage.onObjectArchived(
+        { bucket: 'my-bucket', region: 'us-west1' },
+        () => 42
+      );
+
+      expect(result.__trigger).to.deep.equal({
+        apiVersion: 2,
+        platform: 'gcfv2',
+        labels: {},
+        eventTrigger: {
+          eventType: 'google.cloud.storage.object.v1.archived',
+          resource: 'my-bucket',
+          service: 'storage.googleapis.com',
+        },
+        regions: ['us-west1'],
+      });
+    });
```

Comment:
Can we also test where opt is provided without a `bucket` property?

Reply:
Added tests for each event type tests and the onOperation tests.